RadixWallet/Features/SettingsFeature/DisplayMnemonics/Coordinator/DisplayMnemonics.swift

```diff
@@ -148,13 +149,21 @@ public struct DisplayMnemonics: Sendable, FeatureReducer {
 			switch delegateAction {
 			case .finishedEarly:
 				state.destination = nil
-			case let .finishedImportingMnemonics(_, importedIDs):
+				return .none
+			case let .finishedImportingMnemonics(_, importedIDs, newBDFS):
 				for imported in importedIDs {
 					state.deviceFactorSources[id: imported.factorSourceID]?.imported()
 				}
 				state.destination = nil
+
+				if let newBDFS {
+					return .run { _ in
+						try await factorSourcesClient.saveNewMainBDFS(newBDFS)
+					}
```

Comment:
Seen this already in several places, could it be moved to a single place, probably ImportMnemonic? or it is need first to do additional logic based on each feature before saving it?

Reply:
cleaned up! see recent commits